client/client-multi/client-cpp/CMakeLists.txt

```diff
@@ -204,6 +204,10 @@ if(NOT MSVC)
    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -ggdb")
 endif()
 
+# "no-unused-private-field" is added since in most of cases it is false positive
+list(APPEND KAA_COMPILE_OPTIONS
+                -Wno-unused-private-field)
```

Comment:
>>Does the flag exist in MSVC ?
In MSVC exists "Unused private fields (CA1823)" not sure "-Wno-unused-private-field" works.
I will appreciate, if you can test. I can add "if else" after.

>>And what fields are unused?
While compilation of Events C++ demo on OS X with gcc I got lot of these dummy warnings:
In file included from /Users/vasiliy/Works/CyberVision/Internal/kaa/git-repos/kaa-vnerozin/client/client-multi/client-cpp/impl/Kaa.cpp:21:
In file included from /Users/vasiliy/Works/CyberVision/Internal/kaa/git-repos/kaa-vnerozin/client/client-multi/client-cpp/test/../kaa/KaaClient.hpp:28:
/Users/vasiliy/Works/CyberVision/Internal/kaa/git-repos/kaa-vnerozin/client/client-multi/client-cpp/test/../kaa/profile/ProfileManager.hpp:107:39: warning: 
      private field 'context_' is not used [-Wunused-private-field]
    IKaaClientContext                &context_;

This field is used (if I comment private filed this module will not compile), but warning is exists. We discussed this with @rasendubi and decided to add this warning stopper.